<launch>
  <arg name="CONTROLLER_TYPE" default="auto" />
  <arg name="DEV" default="/dev/input/js0" />
  <arg name="POSE_REMAP" default="joy_pose" />
  <arg name="FRAME_ID" default="my_map" />

  <arg name="gui" default="true" />

  <node pkg="joy" type="joy_node" name="joy_driver" output="screen" >
    <param name="dev" type="string" value="$(arg DEV)" />
    <param name="deadzone" value="0.2" />
    <param name="autorepeat_rate" value="40" />
    <param name="coalesce_interval" value="0.025" />
  </node>
  <node pkg="jsk_teleop_joy" type="joy.py" name="joy" output="screen"
        clear_params="true">
    <param name="controller_type" value="$(arg CONTROLLER_TYPE)" />
    <rosparam subst_value="true">
      plugins:
        'MarkerArray':
          class: 'JoyMarkerArray'
          args:
            menu: "overlay_menu"
            type: 3
            lock_xy: True
            publish_pose: True
            pose: $(arg POSE_REMAP)
            name: "joy_marker_array"
            namespace: "cylinder_array"
            frame_id: $(arg FRAME_ID)
            follow_view: true
            set_marker: "set_marker"
    </rosparam>
  </node>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_teleop_joy)/launch/config/sample_marker_array.rviz">
    </node>
  </group>

</launch>
